### DISCUSSION TITLE
CVE-2021-27291

### DISCUSSION THREAD

**Discussion by BKirov** [argoproj/argo-cd#17671] [General]
Hello , even with the latest version we are having this issue   which is saying : 
`The library pygments version 2.6.1 was detected in Pipenv library manager located at /tmp/_argocd-repo/ee3e564c-1edb-48df-8d64-4ca91099f6f1/project_onboarder/Pipfile.lock and is vulnerable to CVE-2021-27291, which exists in versions >= 1.1, < 2.7.4.

The vulnerability was found in the [Github Security Advisory](https://github.com/advisories/GHSA-pq64-v7f5-gqh8) with vendor severity: High ([NVD](https://nvd.nist.gov/vuln/detail/CVE-2021-27291) severity: High).

This vulnerability has a known exploit available. Source: [Github](https://gist.github.com/b-c-ds/b1a2cc0c68a35c57188575eb496de5ce).

The vulnerability can be remediated by updating the library to version 2.7.4 or higher, using pipenv install --selective-upgrade pygments.`
Are you going to update this library and if yes , when are you planing to do that ?

**Comment by crenshaw-dev** [comment]
Argo CD doesn't run any python at runtime. I suspect the dependency is in our docs build pipeline. It's safe to ignore for running Argo CD environments. 

Even so, feel free to open a PR to bump the dependency in docs/requirements.txt.

**Reply by BKirov** [reply]
Hi @crenshaw-dev  , i am so sorry but can you please explain in docs/requerements.txt is already pygments==2.15.0
 
 but in vulnerability is saying that we need  2.7.4. and higher , which we already have here ? or i am missing something?

**Reply by crenshaw-dev** [reply]
Oh lol, the scanner is picking up a Pipfile from the repo cache. So the old dependency isn't in Argo CD itself, it's in one of your internal repos. Take another look at the file path.

**Comment by BKirov** [comment]
Hi @crenshaw-dev  again , what about this  CVE-2023-46402
` The library github.com/whilp/git-urls version 0.0.0-20191001220047-6db9661140c0 was detected in Golang binary located at /shared/argocd-dex and is vulnerable to CVE-2023-46402, which exists in versions <= 1.0.1.` , 
The problem here is that the latest version of this repo is https://github.com/whilp/git-urls 1.0.0 
So is there any option we can workaround this ?

**Comment by BKirov** [comment]
Hi @crenshaw-dev  , can you please give your opinion here ^^^
